File: 02/cpp/Rhombus.cpp
#include "Rhombus.hpp"

#include <cmath>

static const double pi = std::acos(-1);

Rhombus::Rhombus(double sideLength, radian angle) : 
    Tetragon(sideLength, sideLength, sideLength, sideLength, angle) {}

double Rhombus::surface_area() noexcept { return sideLengths[0] * sideLengths[0] * std::sin(angle); }

double Rhombus::perimeter() noexcept { return 4 * sideLengths[0]; }

File: 02_2017/Shapes/src/main/java/com/dabek/jakub/shapes/Rhombus.java
package com.dabek.jakub.shapes;

public class Rhombus extends Tetragon {

    public Rhombus(double sideLength, double angle) {
        super(sideLength, sideLength, sideLength, sideLength, angle);
    }

    @Override
    public double surfaceArea() {
        return sideLengths[0] * sideLengths[0] * Math.sin(angle);
    }

    @Override
    public double perimeter() {
        return 4 * sideLengths[0];
    }

}